### PR TITLE
Beechwoods fixes

### PR DESCRIPTION
**Pull request by cliu5764**
Fixed the following compiler warnings/errors:
include/zephyr/net/buf.h - casts assignment to prevent warning 
include/zephyr/net/hostname.h - Just prevents a compiler warning due to unused arg. Should use "ARG_UNUSED(<arg>) rather than "(void) <arg>;"
include/zephyr/net/net_context.h - casts and ARG_UNUSED() to prevent compiler warnings.
include/zephyr/net/net_if.h - several ARG_UNUSED().
include/zephyr/net/net_ip.h - cast in an assignment (presumably to prevent warning).
include/zephyr/posix/time.h - cast to prevent warning
include/zephyr/sys/byteorder.h - lots of casts to prevent warnings
include/zephyr/sys/util.h - cast to prevent warnings

### CODE REVIEW

File: include/zephyr/net/hostname.h [hunk 1]
```diff
@@ -103,6 +103,8 @@ int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 static inline int net_hostname_set_postfix(const uint8_t *hostname_postfix,
 					   int postfix_len)
 {
+	(void) hostname_postfix;
```
Comment (thedjnK) [inline include/zephyr/net/hostname.h:106]: zephyr uses ARG_UNUSED() for this

File: include/zephyr/net/net_context.h [hunk 2]
```diff
@@ -435,7 +435,7 @@ static inline void net_context_set_accepting(struct net_context *context,
 	if (accepting) {
 		context->flags |= NET_CONTEXT_ACCEPTING_SOCK;
 	} else {
-		context->flags &= ~NET_CONTEXT_ACCEPTING_SOCK;
+	  context->flags &= (uint16_t)~NET_CONTEXT_ACCEPTING_SOCK;
```
Comment (thedjnK) [inline include/zephyr/net/net_context.h:438]: ??

**Review by thedjnK — CHANGES_REQUESTED** [review]

**Comment by jukkar** [comment]
Some general notes:
* Please do not use merge commits in the PRs.
* See https://docs.zephyrproject.org/latest/contribute/guidelines.html#pull-requests-and-issues for some contribution guidelines. Each commit needs to have clear subject, body and signed-off-by lines
* Split the changes to logical components, all network related fixes to one commit, posix to one, and utility fixes to one commit inside this PR.

**Comment by cliu5764** [comment]
Going to redo it with 3.6 release

**Pull request closed without merging**